### PR TITLE
experiment: Use pathed exports for internal APIs

### PR DESCRIPTION
**Pull request by tylerbutler**
This PR is an experiment in using the exports field to provide an explicit entrypoint to use for internal purposes, separate from the package's public API. I used the devtools packages as a testbed.

The devtools-view package required the most changes. It is an ESM-only package (`type: module` in package.json), so when I switched the moduleResolution to node16, I had to update all the imports to add file extensions. I also had to update the chalk import. This might be alleviated by updating to a more recent version of chalk that is ESM-only, but that will cause other problems.

Switching to CJS by removing the `type: module` setting in package.json alleviates some problems, but causes others, because some of the dependencies are also ESM-only. That would require switching to dynamic imports for those packages, which is a decision best left to the code owners.

I wasn't able to get devtools-view to compile because of a react thing I don't understand. The problem is with a dependency whose most recent version is 3 years old, so the simplest solution might be to find a new dependency.

### CODE REVIEW

File: packages/tools/devtools/devtools-core/package.json [hunk 1]
```diff
@@ -11,14 +11,38 @@
...
-	"types": "dist/index.d.ts",
+	"types": "dist/devtools-core-public.d.ts",
 	"scripts": {
+		"api-extractor:commonjs": "api-extractor run --local",
+		"api-extractor:esnext": "api-extractor run --local --config api-extractor.esnext.json",
 		"build": "fluid-build . --task build",
 		"build:commonjs": "fluid-build . --task commonjs",
 		"build:compile": "fluid-build . --task compile",
-		"build:docs": "api-extractor run --local",
+		"build:docs": "npm run api-extractor:commonjs && npm run api-extractor:esnext",
```
Comment (Josmithr) [inline packages/tools/devtools/devtools-core/package.json:45]: Can these run concurrently?

File: fluidBuild.config.cjs [hunk 2]
```diff
@@ -38,9 +38,11 @@ module.exports = {
 		"build:genver": [],
 		"typetests:gen": ["^tsc", "build:genver"], // we may reexport type from dependent packages, needs to build them first.
 		"tsc": tscDependsOn,
-		"build:esnext": tscDependsOn,
+		"build:esnext": [...tscDependsOn],
```
Comment (Josmithr) [inline fluidBuild.config.cjs:41]: Is this change needed?

Comment (tylerbutler) [reply fluidBuild.config.cjs:41]: Not technically but it makes refactoring the tasks easier. That's what led to it - I was changing the tasks around and this stuck around
